cachito/web/utils.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+from flask import request, url_for
+
+
+def pagination_metadata(pagination_query):
+    """
+    Returns a dictionary containing metadata about the paginated query.
+    This must be run as part of a Flask request.
+    :param pagination_query: flask_sqlalchemy.Pagination object
+    :return: a dictionary containing metadata about the paginated query
+    """
+
+    pagination_data = {
+        'page': pagination_query.page,
+        'per_page': pagination_query.per_page,
+        'total': pagination_query.total,
+        'pages': pagination_query.pages,
+        'first': url_for(
+            request.endpoint, page=1, per_page=pagination_query.per_page, _external=True),
+        'last': url_for(
+            request.endpoint, page=pagination_query.pages, per_page=pagination_query.per_page,
+            _external=True)
+    }
+
+    if pagination_query.has_prev:
+        pagination_data['prev'] = url_for(request.endpoint, page=pagination_query.prev_num,
+                                          per_page=pagination_query.per_page, _external=True)
+    if pagination_query.has_next:
+        pagination_data['next'] = url_for(request.endpoint, page=pagination_query.next_num,
```

Comment:
Nitpick: Could you fix the indentation style so it matches the rest of the code?